torchdata/utils/pin_memory.py

```diff
@@ -0,0 +1,34 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import collections
+
+import torch
+
+
+def pin_memory_fn(data, device=None):
```

Comment:
Non-blocking: A short docstring and add this to `torchdata.datapipes.utils.rst` will be nice.

Reply:
Good point!